src/kernel-install/kernel-install.in

```diff
@@ -297,6 +297,14 @@ trap cleanup EXIT
 
 KERNEL_INSTALL_STAGING_AREA="$(mktemp -d -t kernel-install.staging.XXXXXXX)"
 
+export KERNEL_INSTALL_IMAGE_TYPE=""
+if [ -f "$1" ]; then
+    KERNEL_INSTALL_IMAGE_TYPE="$(bootctl kernel-identify "$1")"
+    if [ "$KERNEL_INSTALL_IMAGE_TYPE" = "uki" ] && [ "$layout" = "" ]; then
```

Comment:
Switching the layout if none is set might break backward-compat. @poettering mentioned an `auto` layout above. Maybe something like 
```suggestion
    if [ "$KERNEL_INSTALL_IMAGE_TYPE" = "uki" ] && [ "$layout" = "auto" ]; then
```
and switching the default to `auto`, would make sense, since that should allow people to still not set anything and keep the current behaviour.

Reply:
Ok, and then eventually pass "auto" to the plugins?  Or should `kernel-install` decide what mode to pick then?  I think the latter would be better.

Reply:
I agree. Otherwise every plugin would need to reimplement that logic and that would only lead to problems down the line.